cmd/deploy.go

```diff
@@ -199,58 +251,95 @@ func runDeploy(cmd *cobra.Command, _ []string, newClient ClientFactory) (err err
 
 	// Default Client Registry, Function Registry or explicit Image required
 	if client.Registry() == "" && f.Registry == "" && f.Image == "" {
-		return ErrRegistryRequired
-	}
+		if interactiveTerminal() {
+			// to be consistent, this should throw an error, with the registry
+			// prompting code placed within config.Prompt and triggered with --confirm
+			fmt.Println("A registry for function images is required. For example, 'docker.io/tigerteam'.")
+			if err = survey.AskOne(
+				&survey.Input{Message: "Registry for function images:"},
+				&config.Registry, survey.WithValidator(
+					NewRegistryValidator(config.Path))); err != nil {
+				return ErrRegistryRequired
+			}
+			fmt.Println("Note: building a function the first time will take longer than subsequent builds")
```

Comment:
Noting the same here as in my previous comment about this text. I also don't really love the little bits of duplicated code between build and deploy, but haven't looked too closely at that.